include/nccl_ofi.h

```diff
@@ -327,7 +326,8 @@ struct nccl_net_ofi_device {
 	 * implementation of get_domain() and should not be called
 	 * from the more general case.
 	 */
-	nccl_net_ofi_domain_t *(*create_domain)(nccl_net_ofi_device_t *dev);
+	nccl_net_ofi_domain_t *(*create_domain)(nccl_net_ofi_device_t *dev,
+						long creating_thread_id);
```

Comment:
Not sure I understand why we need this change.  I'd rather minimize the number of places that are making choices based on thread ids as much as possible.  THis seems to go the opposite direction.

Reply:
Tried to address this in my latest force push with a new commit that adds device helper function "nccl_net_ofi_device_remove_domain" that iterates through the `domain_table` map as you suggested, and then removed `creating_thread_id` as a domain member altogether.